assembly/src/tokens/stream.rs

```diff
@@ -25,24 +26,40 @@ impl<'a> TokenStream<'a> {
             return Err(ParsingError::empty_source());
         }
         let mut tokens = Vec::new();
-        let mut doc_comments = BTreeMap::new();
+        let mut proc_comments = BTreeMap::new();
+        let mut module_comment = None;
 
-        let mut comment = Comment(None);
+        let mut comment_builder = CommentBuilder(None);
 
         for line in source.lines() {
             let line = line.trim();
             if line.starts_with(DOC_COMMENT_PREFIX) {
-                comment.append_line(line);
-            } else if line.is_empty() && !comment.is_empty() {
-                return Err(ParsingError::malformed_doc_comment(tokens.len()));
+                comment_builder.append_line(line);
+            } else if line.is_empty()
+                && !comment_builder.is_empty()
+                && tokens.is_empty()
+                && module_comment.is_none()
+            {
+                // If we haven't read any tokens yet, but already have built a comment, a new line
+                // must indicate the end of a module comment.
+                module_comment = comment_builder.take_comment();
             } else {
```

Comment:
We should be getting into this `else` clause only if the line starts with a token. So, blank lines, lines which start with `#` and `#!` should be filtered out before we get here.

Most of this is already working:
- `#!` lines are already being filtered out.
- blank lines should be filtered out once we update the `else if line.is_empty()` clause as I suggested above.
- We need to add a clause for filtering out `#!` lines.

Once this is done, you should be able to get rid of `!line_tokens.is_empty()` check on line 54.